src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -590,6 +590,17 @@ def load_checkpoint(
     def register_strategies(cls, strategy_registry: _StrategyRegistry) -> None:
         strategy_registry.register("xla_fsdp", cls, description=cls.__class__.__name__)
 
+    def _parse_fsdp_kwargs(self) -> Dict:
+        # this needs to be delayed because `self.precision` isn't available at init
+        kwargs = copy(self._fsdp_kwargs)
```

Comment:
Micro nit: if a shallow copy is sufficient, you could save the import and do `self._fsdp_kwargs.copy()`